lib/ui/widgets/alarm_tile.dart

```diff
@@ -1,88 +1,83 @@
   }) : super(key: key);
 
-
   @override
   State<AlarmTile> createState() => _AlarmTileState();
 }
 
 class _AlarmTileState extends State<AlarmTile> {
-
   @override
   Widget build(BuildContext context) {
-
-    if (widget.isDeleted) return const SizedBox(); // gibt es eine bessere Lösung??
-
-    String hour = widget.time.hour < 10 ? '0${widget.time.hour}' : widget.time
-        .hour.toString();
-    String minute = widget.time.minute < 10 ? '0${widget.time.minute}' : widget
-        .time.minute.toString();
+    final hour = widget.time.hour < 10 ? '0${widget.time.hour}' : widget.time.hour.toString();
+    final minute =
+        widget.time.minute < 10 ? '0${widget.time.minute}' : widget.time.minute.toString();
 
     return Padding(
       padding: const EdgeInsets.only(top: 8.0),
       child: Card(
-        margin: const EdgeInsets.fromLTRB(20.0, 6.0, 20.0, 0.0),
+        margin: const EdgeInsets.fromLTRB(
+          20.0,
+          6.0,
+          20.0,
+          0.0,
+        ),
         child: ListTile(
           leading: Text(
-            '$hour : $minute', style: const TextStyle(fontSize: 25.0),),
-          title: Text(widget.name, style: const TextStyle(fontSize: 20.0),),
-          subtitle: Text(widget.rep ? 'Daily' : 'Once',
-            style: const TextStyle(fontSize: 20.0),),
+            '$hour : $minute',
+            style: const TextStyle(fontSize: 25.0),
+          ),
+          title: Text(
+            widget.name,
+            style: const TextStyle(fontSize: 20.0),
+          ),
+          subtitle: Text(
+            widget.rep ? 'Daily' : 'Once',
+            style: const TextStyle(fontSize: 20.0),
+          ),
           trailing: Switch(
             value: widget.isOn,
             onChanged: (val) {
-              setState(() {
-                widget.isOn = val;
-                alarms[widget.index].changeAlarm(val);
-              });
+              widget.onAlarmStatusChanged(val);
             },
             activeColor: Colors.black,
           ),
-          onTap: () {
-            _pushChangeAlarmScreen(context);
+          onTap: () async {
+            await _pushChangeAlarmScreen(context);
           },
         ),
       ),
     );
   }
 
   Future<void> _pushChangeAlarmScreen(BuildContext context) async {
-    final result = await Navigator.pushNamed(
-        context,
-        '/change_alarm',
-        arguments: {'index': widget.index}
+    await Navigator.pushNamed(
+      context,
+      '/change_alarm',
+      arguments: {
+        'index': widget.index,
+      },
     );
-    if (result == 'Yes') {
-      setState(() {
-        widget.name = alarms[widget.index].name;
-        widget.isOn = alarms[widget.index].isOn;
-        widget.time = alarms[widget.index].time;
-        widget.time = alarms[widget.index].time;
-      });
-    }
-    if (result == 'Deleted') {
-      setState(() {
-        widget.isDeleted = true;
-      });
-    }
+    widget.onUpdate();
   }
```

Comment:
Hier kann der Code reduziert werden, dass nachdem ein Alarm geändert wurde einfach der onUpdate() Callback aufgerufen wird, wodurch der übergeordnete AlarmClockScreen weiß, dass sich etwas geändert hat in einem seiner Alarme